### ISSUE TITLE
Visualize a Spatial Dataset on SAP HANA Database in SAP HANA Cloud

### ISSUE THREAD

**Issue by jastill** [SAPDocuments/Issues#676]
Tutorials: https://developers.sap.com/tutorials/hana-cloud-trial-qgis-2.html
--------------------------

Write here how you think we can improve the tutorial ...

In step 2 when the layer is added to the DBADMIN schema in QGIS the following error occurs

------------------------
Failed to import some layers!

Feature write errors:
Creation error for features from #-9223372036854775808 to #-9223372036854775808. Provider errors was: 
Failed to add features: General error;274 inserted value too large for column: Failed in "fclass" column with the value 'Kindertagesst�tte Pittiplatsch'

Only 0 of 100775 features written.
-------------------------

The value in the error is shown in the "name" column of the data in QGIS. It seems that in the import, it appears to attempt to insert the value into the fclass field.

I am running QGIS 3.18.3-Zurich in Crostini with hanaclient-latest.... installed tonight from tools.hana.ondemand.com

The table is created in HANA Cloud, but the data is not inserted.

Regards
John

**Comment by VijayKrishnanSR** [comment]
Hi John,

Thank you for letting me know about the issue. I have forwarded this to our product experts. I will get back to you soon.

Regards,
Vijay

**Comment by mkemeter** [comment]
Hi John,

many thanks for reporting this issue! It seems that you have encountered a bug, that has been fixed from QGIS 3.20 onwards:
https://github.com/qgis/QGIS/commit/85b1a87c8ad44001b4b67d4299a6d429f1cde825

To verify that the current version does the job, I have upgraded my QGIS to 3.22 and tried the latest download from Geofabrik. The import was successful. 

Let me know, if the latest version of QGIS (or at least > 3.20) works for you too.

Regards,
Mathias
